scripts/linters/general_purpose_linter.py

```diff
@@ -727,23 +757,29 @@ def _check_bad_patterns(self):
                             total_error_count += 1
 
                 for regexp in BAD_PATTERNS_REGEXP:
-                    if check_bad_pattern_in_file(
-                            filepath, file_content, regexp):
-                        failed = True
+                    tmp_failed, bad_pattern_summary_messages = (
+                        check_bad_pattern_in_file(
+                            filepath, file_content, regexp))
+                    if tmp_failed:
+                        summary_messages.extend(bad_pattern_summary_messages)
                         total_error_count += 1
 
-                temp_failed, temp_count = check_file_type_specific_bad_pattern(
-                    filepath, file_content)
-                failed = failed or temp_failed
+                temp_failed, temp_count, bad_pattern_summary_messages = (
+                    check_file_type_specific_bad_pattern(
+                        filepath, file_content))
+                failed = failed or temp_failed or tmp_failed
```

Comment:
This is confusing please add a better name for `temp and tmp`

Reply:
Done.